### PR TITLE
Remove v3 api form SSA

### PR DESCRIPTION
**Pull request by borod108**
Remove the usage of the ovirt gem and version 3 api from the SSA

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1535782

Depends on: https://github.com/ManageIQ/manageiq-smartstate/pull/98

### CODE REVIEW

**Comment by borod108** [comment]
@agrare what say you?

File: app/models/manageiq/providers/redhat/inventory/collector.rb [hunk 1]
```diff
@@ -157,4 +157,40 @@ def collect_dc_domains(dc)
...
+      disk_attachments = connection.follow_link(vm.disk_attachments)
+      disk_attachments.collect do |disk_attachment|
+        connection.follow_link(disk_attachment.disk)
+      end
+    end
+  end
+
+  def vm_or_template_by_path(path)
+    uuid = ::File.basename(path, '.*')
+    vm      = vm_by_uuid(uuid)
```
Comment (agrare) [inline app/models/manageiq/providers/redhat/inventory/collector.rb]: Lots of extra spaces here, did one of these variables used to have a longer name?

File: app/models/manageiq/providers/redhat/infra_manager/vm_or_template_shared/scanning.rb [hunk 2]
```diff
@@ -151,7 +151,7 @@ def connect_to_ems(ost)
     $log.info "#{log_header}: Connecting to [#{ems_display_text}] for VM:[#{@vm_cfg_file}]"
 
     begin
-      ost.miqRhevm = ext_management_system.rhevm_inventory(:force_v3 => true)
+      ost.miqRhevm = ManageIQ::Providers::Redhat::Inventory::Collector.new(ext_management_system, nil)
```
Comment (agrare) [inline app/models/manageiq/providers/redhat/infra_manager/vm_or_template_shared/scanning.rb]: Not for this, but I think we should make the Collector initializer `initialize(manager, target = nil)` in core if it isn't required.

**Comment by agrare** [comment]
@borod108 these both say they depend on each other, if I had to guess I'd say this has to be merged first is that right?

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit 33059244eec76754dba2632c0832fa2645276542
Author: Adam Grare <agrare@redhat.com>
Date:   Wed Aug 14 16:54:40 2019 -0400

    Merge pull request #394 from borod108/bugs/1535782smart_no_v3
    
    Remove v3 api form SSA
    
    (cherry picked from commit 343edb5d474accae5c9fe7c1af33fe4a7eba7bcc)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1535782
```

**Pull request merged**